test/APIConsumer_test.js

```diff
@@ -1,29 +1,55 @@
-const { expect } = require('chai')
+const chai = require('chai')
+const {expect} = require('chai')
+const BN = require('bn.js');
+chai.use(require('chai-bn')(BN));
 
 describe('APIConsumer', async function () {
+  
   let apiConsumer, mockOracle, linkToken
 
   beforeEach(async () => {
-    await deployments.fixture(['mocks', 'api'])
+
+    //if local chain, then we use mocks
+    if (await getChainId() == 31337) {
+      await deployments.fixture(['mocks', 'api'])
+      const LinkToken = await deployments.get('LinkToken')
+      linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
+      const MockOracle = await deployments.get('MockOracle')
+      mockOracle = await ethers.getContractAt('MockOracle', MockOracle.address)
+    }
 
     // Then, we can get the contracts like normal
-    const LinkToken = await deployments.get('LinkToken')
-    linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
     const APIConsumer = await deployments.get('APIConsumer')
     apiConsumer = await ethers.getContractAt('APIConsumer', APIConsumer.address)
-    const MockOracle = await deployments.get('MockOracle')
-    mockOracle = await ethers.getContractAt('MockOracle', MockOracle.address)
+      
   })
 
   it('Should successfully make an external API request', async () => {
-    const expected = '777'
-    await linkToken.transfer(apiConsumer.address, '2000000000000000000')
-    const transaction = await apiConsumer.requestVolumeData()
-    const tx_receipt = await transaction.wait()
-    const requestId = tx_receipt.events[0].topics[1]
-    const returnData = web3.utils.padLeft(web3.utils.padLeft(web3.utils.toHex(expected)), 64)
-    const tx = await mockOracle.fulfillOracleRequest(requestId, returnData)
-    await tx.wait()
-    expect(await apiConsumer.volume()).to.equal(expected)
+    // If local chain, do tests using mocks
+    if (await getChainId() == 31337) {
+      const expected = '777'
+      await linkToken.transfer(apiConsumer.address, '2000000000000000000')
+      const transaction = await apiConsumer.requestVolumeData()
+      const tx_receipt = await transaction.wait()
+      const requestId = tx_receipt.events[0].topics[1]
+      const returnData = web3.utils.padLeft(web3.utils.padLeft(web3.utils.toHex(expected)), 64)
+      const tx = await mockOracle.fulfillOracleRequest(requestId, returnData)
```

Comment:
could we just have the if here?

ie: if local -> call fulfilment. Right now, we basically have to write two totally different tests for local and testnet, when its really only 1 function that we need to call.

Reply:
have gotten rid of the if/else, and just have if as requested. The only problem is, once again you can't have 2 different tests in this case, you can't compare the local result to the expected variable, but then for the public network check that its > 0. So to make it more uniform, both scenarios now just check for > 0 result